### PR TITLE
Fix a warning. Followup to #13591

### PR DESCRIPTION
**Pull request by kripken**

### CODE REVIEW

File: src/library_pthread.js [hunk 1]
```diff
@@ -479,9 +479,12 @@ var LibraryPThread = {
         + '\nIf you want to throw an explicit error instead of the risk of deadlocking in those cases, use setting `-s PTHREAD_POOL_SIZE_STRICT=2`.'
 #endif
         );
-#else
-        err('No available workers in the PThread pool');
-#endif
+#else // PTHREAD_POOL_SIZE_STRICT
+#if ASSERTIONS
```
Comment (RReverser) [inline src/library_pthread.js:483]: Seems like a duplication, the branch before `else` is also for `ASSERTIONS`.

Comment (kripken) [reply src/library_pthread.js]: Thanks, all the ifdefs confused me and I matched the wrong ifs with elses and ends...

Comment (RReverser) [reply src/library_pthread.js]: I know the feeling. Another reason to switch to regular JS ifs :)

**Review by RReverser — APPROVED** [review]
I think we still should print an error message at least for `PTHREAD_POOL_SIZE_STRICT == 2` mode, since a lot of people are probably ignoring the return value of `pthread_create`, but we can probably re-add this later if necessary.

For now this behaviour is good enough to unblock the release.

**Pull request merged**